packages/grafana-ui/src/components/TransformersUI/CalculateFieldTransformerEditor.tsx

```diff
@@ -0,0 +1,203 @@
+    });
+  };
+
+  onToggleReplaceFields = (evt: ChangeEvent<HTMLInputElement>) => {
+    const { options } = this.props;
+    this.props.onChange({
+      ...options,
+      replaceFields: !options.replaceFields,
+    });
+  };
+
+  onAliasChanged = (evt: ChangeEvent<HTMLInputElement>) => {
+    const { options } = this.props;
+    this.props.onChange({
+      ...options,
+      alias: evt.target.value,
+    });
+  };
+
+  onStatsChange = (stats: string[]) => {
+    this.props.onChange({
+      ...this.props.options,
+      reducer: stats.length ? (stats[0] as ReducerID) : ReducerID.sum,
+    });
+  };
+
+  render() {
+    const { options } = this.props;
+    const { names, selected } = this.state;
+    const reducer = fieldReducers.get(options.reducer);
+    return (
+      <div>
+        <Label>Numeric Fields</Label>
+        <InlineList
+          items={names}
+          renderItem={(o, i) => {
+            return (
+              <span
+                className={css`
+                  margin-right: ${i === names.length - 1 ? '0' : '10px'};
+                `}
+              >
+                <FilterPill
+                  onClick={() => {
+                    this.onFieldToggle(o);
+                  }}
+                  label={o}
+                  selected={selected.indexOf(o) > -1}
+                />
+              </span>
+            );
+          }}
+        />
+        <Label>Calculation</Label>
+        <StatsPicker stats={[options.reducer]} onChange={this.onStatsChange} defaultStat={ReducerID.sum} />
+        <Label>Alias</Label>
+        <Input value={options.alias} placeholder={reducer.name} onChange={this.onAliasChanged} />
+
+        <Label>Replace all fields</Label>
+        <Switch checked={options.replaceFields} onChange={this.onToggleReplaceFields} />
+
+        {/*  nullValueMode?: NullValueMode; */}
+      </div>
+    );
+  }
+}
+
+interface FilterPillProps {
+  selected: boolean;
+  label: string;
+  onClick: React.MouseEventHandler<HTMLElement>;
+}
+const FilterPill: React.FC<FilterPillProps> = ({ label, selected, onClick }) => {
+  const theme = useContext(ThemeContext);
+  return (
+    <div
+      className={css`
+        padding: ${theme.spacing.xxs} ${theme.spacing.sm};
+        color: white;
+        background: ${selected ? theme.palette.blue95 : theme.palette.blue77};
+        border-radius: 16px;
+        display: inline-block;
+        cursor: pointer;
+      `}
+      onClick={onClick}
+    >
+      {selected && (
+        <Icon
+          className={css`
+            margin-right: 4px;
+          `}
+          name="check"
+        />
+      )}
+      {label}
+    </div>
+  );
+};
+
+export const calculateFieldTransformRegistryItem: TransformerRegistyItem<CalculateFieldTransformerOptions> = {
+  id: DataTransformerID.calculateField,
+  editor: CalculateFieldTransformerEditor,
+  transformation: standardTransformers.calculateFieldTransformer,
+  name: 'Calculate Field',
```

Comment:
Naming is hard... Maybe `Create dynamic field?`

Reply:
Add field from calculation?

Reply:
Add field from calculation sounds good. At least it clearly describes that a new field will be created 👍